### PR TITLE
Add switch to Prebid permutive segments

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Add a switch + tests for Permutive Audience Connector in Prebid.

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: common/app/conf/switches/CommercialSwitches.scala [hunk 1]
```diff
@@ -119,7 +119,7 @@ trait CommercialSwitches {
   val PermutiveSwitch = Switch(
     Commercial,
     "permutive",
-    "Enable Permutive library loading",
+    "Enable Permutive library loading. See ",
```
Comment (mxdvl) [inline common/app/conf/switches/CommercialSwitches.scala:122]:
```suggestion
    "Enable Permutive library loading.",
```

File: common/app/model/dotcomrendering/pageElements/PageElement.scala [hunk 2]
```diff
@@ -236,18 +236,12 @@ object GenericAtomBlockElement {
   implicit val GenericAtomBlockElementWrites: Writes[GenericAtomBlockElement] = Json.writes[GenericAtomBlockElement]
 }
 
-case class GuideAtomBlockElementItem(title: Option[String], body: String)
```
Comment (arelra) [inline common/app/model/dotcomrendering/pageElements/PageElement.scala:239]: Not sure how this is related to the prebid permutive change?

Comment (mxdvl) [reply common/app/model/dotcomrendering/pageElements/PageElement.scala:239]: What happened there, I don't know!

File: static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.spec.js [hunk 3]
```diff
@@ -164,109 +169,138 @@ describe('initialise', () => {
...
+
+    test.each([
+        [true, false],
+        [false, true],
+        [false, false],
+    ])('should not generate RTD when Permutive is %s and prebidPermutiveAudience is %s', (p, a) => {
+        config.set('switches.permutive', p);
+		config.set('switches.prebidPermutiveAudience', a);
+		prebid.initialise(window);
+        const rtcData = window.pbjs.getConfig('realTimeData');
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.spec.js:194]: Some alignment issues here?

File: static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.spec.js [hunk 4]
```diff
@@ -164,109 +169,138 @@ describe('initialise', () => {
...
+            expect(getAdvertById).toHaveBeenCalledWith('foo');
+            expect(dummyAdvert.size).toMatchObject([100, 100]);
+            expect(dummyAdvert.hasPrebidSize).toBe(true);
         });
 
-        prebid.initialise(window);
-
-        expect(bidWonEventName).toBe('bidWon');
-        expect(window.pbjs.onEvent).toHaveBeenCalledTimes(1);
+        test('should not respond if height missing from prebid data', () => {
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.spec.js:233]: Could you potentially macro the three 'should not respond' tests: height, width, adUnitCode?
On a quick a look it looks like the only difference is the object returned from bidWonEventHandler ?

**Review by arelra — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.spec.js (hunk 4 above)
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.spec.js:233]: Good shout.

**Pull request merged**